rc/vue.kak

```diff
@@ -0,0 +1,88 @@
+
+# http://w3.org/html
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+# Detection
+# ‾‾‾‾‾‾‾‾‾
+
+hook global BufCreate .*\.vue %{
+    set-option buffer filetype vue
+}
+
+# Initialization
+# ‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+
+hook global WinSetOption filetype=vue %{
+    require-module vue
+
+#     hook window ModeChange pop:insert:.* -group "%val{hook_param_capture_1}-trim-indent"  html-trim-indent
+#     hook window InsertChar '>' -group "%val{hook_param_capture_1}-indent" html-indent-on-greater-than
+#     hook window InsertChar \n -group "%val{hook_param_capture_1}-indent" html-indent-on-new-line
+# 
+    # hook -once -always window WinSetOption "filetype=.*" "
+        # remove-hooks window ""%val{hook_param_capture_1}-.+""
+    # "
+}
+
+hook -group html-highlight global WinSetOption filetype=vue %{
+#    add-highlighter "window/%val{hook_param_capture_1}" ref html
+#    hook -once -always window WinSetOption "filetype=.*" "
+#        remove-highlighter ""window/%val{hook_param_capture_1}""
+#    "
+}
+
+
+provide-module vue %[
+
+try %{
+    require-module css
+    require-module javascript
+}
+
+# Highlighters
+# ‾‾‾‾‾‾‾‾‾‾‾‾
+
+add-highlighter shared/html regions
```

Comment:
These highlighters and the commands below will clash with the ones from html.kak.
This happens if both a Vue file and an HTML file are edited in the same session.

This one should probably be renamed to `shared/vue`

Reply:
@krobelus thanks for your feedback, tbh it's not even working at this point, I have zero knowledge on the syntax that's being used here. For now I'd just like to make it highlight the file as if it was a standard html document, but have it have a type of an HTML file. How would I do that with minimal effort + without copying + pasting the whole html file :D

Reply:
The easiest way is probably to not change anything, and just use the `html` filetype for `*.vue` files.

I guess your main motivation for this is to be able to use a Vue language server
(https://github.com/kak-lsp/kak-lsp/issues/442).
If that's all, then why not use something like

```toml
[language.html]
filetypes = ["html"]
roots = [".git"]
command = "vue-language-server" # was html-language-server
```

in your `~/.config/kak-lsp/kak-lsp.toml`? Assuming you don't need the html-language-server.

Alternatively, copy/pasting and replacing `html` with `vue`  isn't wrong either. It's not beautiful but it works.
The `c-family.kak` script in Kakoune supports multiple related filetypes, that might be relevant.

Reply:
... man you are genious